.github/workflows/ci.yml

```diff
@@ -42,370 +42,70 @@ jobs:
-      - name: Checkout Pex
-        uses: actions/checkout@v3
-        with:
-          # We need branches and tags for some ITs.
-          fetch-depth: 0
-          path: repo
-      - name: Setup Python ${{ join(matrix.python-version, '.') }}
-        uses: actions/setup-python@v4
-        with:
-          python-version: "${{ join(matrix.python-version, '.') }}"
-      - name: Expose Pythons
-        uses: pantsbuild/actions/expose-pythons@e63d2d0e3c339bdffbe5e51e7c39550e3bc527bb
-        with:
-          skip: "${{ steps.calculate-pythons-to-expose.outputs.skip }}"
-      - name: Cache Pyenv Interpreters
-        uses: actions/cache@v3
-        with:
-          path: ${{ env._PEX_TEST_DEV_ROOT }}/pyenv
-          key: ${{ matrix.os }}-${{ runner.arch }}-pex-test-dev-root-pyenv-v1
-      - name: Cache Devpi Server
-        uses: actions/cache@v3
-        with:
-          path: ${{ env._PEX_TEST_DEV_ROOT }}/devpi
-          # We're using a key suffix / restore-keys prefix trick here to get an updatable cache.
-          # See: https://github.com/actions/cache/blob/main/tips-and-workarounds.md#update-a-cache
-          key: ${{ matrix.os }}-${{ runner.arch }}-${{ env._TOX_ENV }}-pex-test-dev-root-devpi-v1-${{ github.run_id }}
-          restore-keys: ${{ matrix.os }}-${{ runner.arch }}-${{ env._TOX_ENV }}-pex-test-dev-root-devpi-v1i Cache
-      - name: Setup SSH Agent
-        uses: webfactory/ssh-agent@v0.5.4
-        env:
-          SSH_PRIVATE_KEY: ${{ secrets.SSH_PRIVATE_KEY }}
-        if: env.SSH_PRIVATE_KEY != ''
-        with:
-          ssh-private-key: ${{ env.SSH_PRIVATE_KEY }}
-      - name: Run Integration Tests
-        uses: pantsbuild/actions/run-tox@b16b9cf47cd566acfe217b1dafc5b452e27e6fd7
-        with:
-          path: repo/tox.ini
-          python: ${{ matrix.tox-env-python }}
-          tox-env: ${{ env._TOX_ENV }} -- ${{ env._PEX_TEST_POS_ARGS }}
-  cpython-integration-tests-legacy:
-    name: (${{ matrix.os }}) Pip ${{ matrix.pip-version }} TOXENV=py${{ matrix.python-version[0] }}${{ matrix.python-version[1] }}-integration
-    needs: org-check
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-22.04
     strategy:
       matrix:
-        include:
-          - os: macos-12
-            python-version: [ 2, 7, 18 ]
-            pip-version: 20
-          - os: ubuntu-22.04
-            python-version: [ 2, 7, 18 ]
-            pip-version: 20
-          - os: ubuntu-22.04
-            python-version: [ 3, 7, 17 ]
-            pip-version: 22_3_1
-          - os: ubuntu-22.04
-            python-version: [ 3, 7, 17 ]
-            pip-version: 23_1_2
-    env:
-      _TOX_ENV: py${{ matrix.python-version[0] }}${{ matrix.python-version[1] }}-pip${{ matrix.pip-version }}-integration
+        tox-env:
+          - py27-pip20
+          - py35-pip20
+          - py311-pip20
+          - py311-pip22_3_1
+          - py311-pip23_1_2
+          - py312-pip23_2
+          - pypy310-pip20
```

Comment:
I dropped pypy27 testing as a pragmatic / cowardly decision. The base image pypy runs into an OpenSSL linking issue in 1 integration test and after much effort I gave up. Unlike CPython 2.7, I'm pretty sure folks have moved on from PyPy 2.7. I'm guessing there are no support contracts for the latter like there are for the former.